proto/public/uber/cadence/api/v1/service_domain.proto

```diff
@@ -0,0 +1,129 @@
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+syntax = "proto3";
+
+package uber.cadence.api.v1;
+
+option go_package = "apiv1";
+option java_multiple_files = true;
+option java_outer_classname = "ApiProto";
+option java_package = "com.uber.cadence.api.v1";
+
+import "google/protobuf/duration.proto";
+import "dependencies/gogoproto.proto";
+import "uber/cadence/api/v1/domain.proto";
+
+service DomainAPI {
+  // RegisterDomain creates a new domain which can be used as a container for all resources.  Domain is a top level
+  // entity within Cadence, used as a container for all resources like workflow executions, task lists, etc.  Domain
+  // acts as a sandbox and provides isolation for all resources within the domain.  All resources belongs to exactly one
+  // domain.
+  rpc RegisterDomain(RegisterDomainRequest) returns (RegisterDomainResponse);
+
+  // DescribeDomain returns the information and configuration for a registered domain.
+  rpc DescribeDomain(DescribeDomainRequest) returns (DescribeDomainResponse);
+
+  // ListDomains returns the information and configuration for all domains.
+  rpc ListDomains(ListDomainsRequest) returns (ListDomainsResponse);
+
+  // UpdateDomain is used to update the information and configuration for a registered domain.
+  rpc UpdateDomain(UpdateDomainRequest) returns (UpdateDomainResponse);
+
+  // DeprecateDomain us used to update status of a registered domain to DEPRECATED.  Once the domain is deprecated
+  // it cannot be used to start new workflow executions.  Existing workflow executions will continue to run on
+  // deprecated domains.
+  rpc DeprecateDomain(DeprecateDomainRequest) returns (DeprecateDomainResponse);
+}
+
+message RegisterDomainRequest {
+  string name = 1;
+  string description = 2;
+  string owner_email = 3;
+  google.protobuf.Duration workflow_execution_retention_period = 4 [(gogoproto.stdduration) = true];
+  bool emit_metric = 5;
+  repeated ClusterReplicationConfiguration clusters = 6;
+  string active_cluster_name = 7;
+  map<string, string> data = 8;
+  string security_token = 9;
+  bool is_global_domain = 10;
+  ArchivalStatus history_archival_status = 11;
+  string history_archival_uri = 12;
+  ArchivalStatus visibility_archival_status = 13;
+  string visibility_archival_uri = 14;
+}
+
+message RegisterDomainResponse {
+}
+
+message DescribeDomainRequest {
+  string name = 1;
+  string id = 2;
+}
```

Comment:
not necessarily worthwhile, but: I'm guessing these are mutually exclusive?  may be worth mentioning optional / use `oneof`.

Reply:
These are indeed mutually exclusive fields. Updated to use oneof.